smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/utils/AuthUtils.kt

```diff
@@ -0,0 +1,39 @@
+package software.amazon.smithy.swift.codegen.utils
+
+import software.amazon.smithy.model.knowledge.ServiceIndex
+import software.amazon.smithy.model.traits.HttpBearerAuthTrait
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.config.DefaultProvider
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.swiftmodules.SmithyHTTPAuthTypes
+
+open class AuthUtils(
+    private val ctx: ProtocolGenerator.GenerationContext
+) {
+    val authSchemesDefaultProvider = DefaultProvider(
+        { getModeledAuthSchemesSupportedBySDK(ctx, it) },
+        isThrowable = false,
+        isAsync = false
+    )
+
+    fun getModeledAuthSchemesSupportedBySDK(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+    ): String {
+        val effectiveAuthSchemes = ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service)
+        var authSchemeList = arrayOf<String>()
```

Comment:
`Lists` are usually preferred in Kotlin over arrays since they are a lot more powerful. It would probably be more idiomatic to use `mutableListOf<String>()` here and update related use cases accordingly

Reply:
Changed as suggested.